File: Python/443. String Compression.py
from typing import List


class Solution:
    # noinspection PyMethodMayBeStatic
    def compress(self, chars: List[str]) -> int:
        length = len(chars)
        index = 0

        i = 0
        while i < length:
            current = chars[i]
            count = 0

            while i < length and chars[i] == current:
                count += 1
                i += 1

            chars[index] = current
            index += 1

            if count > 1:
                for char in str(count):
                    chars[index] = char
                    index += 1

        return index

File: PHP/443. String Compression.php
<?php

/** @noinspection PhpMultipleClassDeclarationsInspection */
class Solution
{
    /**
     * @param  string[]  $chars
     *
     * @noinspection PhpUnused
     */
    public function compress(array &$chars): int
    {
        $length = count($chars);
        $index = 0;

        for ($i = 0; $i < $length;) {
            $currentChar = $chars[$i];
            $count = 0;

            while ($i < $length && $chars[$i] === $currentChar) {
                $count++;
                $i++;
            }

            $chars[$index++] = $currentChar;

            if ($count > 1) {
                foreach (str_split($count) as $char) {
                    $chars[$index++] = $char;
                }
            }
        }

        return $index;
    }
}
